server/lib/oauth2-client/oauth2.executeOauth2HTTPQuery.js

```diff
@@ -0,0 +1,140 @@
+const queryString = require('query-string');
+const { ClientCredentials } = require('simple-oauth2');
+const { default: axios } = require('axios');
+const logger = require('../../utils/logger');
+const { OAUTH2 } = require('../../utils/constants');
+
+/**
+ * @description Refresh Oauth2 token in case of it expired .
+ * @param {Object} variable - Gladys variable manager.
+ * @param {string} serviceId - Gladys service id call method.
+ * @param {string} userId - Gladys userId of current session.
+ * @returns {Object} Response of oauth2 refresh token query .
+ * @example
+ * oauth2.refreshTokenAccess('7fdsf4s68r4gfr68f4r63csd7f6f4c3r85',
+ *    {...}, '78v4f3df83g74v1fsd8375f63gvrf5c');
+ */
+async function refreshTokenAccess(variable, serviceId, userId) {
+  const clientId = await variable.getValue(OAUTH2.VARIABLE.CLIENT_ID, serviceId, userId);
+  const secret = await variable.getValue(OAUTH2.VARIABLE.CLIENT_SECRET, serviceId, userId);
+  const accessToken = JSON.parse(await variable.getValue(OAUTH2.VARIABLE.ACCESS_TOKEN, serviceId, userId));
```

Comment:
why do you save the access_token as a JSON ? not just the value ?

Reply:
Because aouth2 access_token contan more than 1 parameter so I store like string al json to pass directly this value when I need authenticate for execute query

complete access token is like this:
{"userid":"xxxxxx","access_token":"2dfexxxxxx7bb2d6f708dddf0d0b3xxxxx7","refresh_token":"44e25xxxxx46b17934112f3ab2xxxxxxcecd0","scope":"user.info,user.metrics,user.activity,user.sleepevents","expires_in":10800,"token_type":"Bearer","expires_at":"2022-04-09T21:57:49.068Z"}